filesystem/src/test/java/com/nytimes/android/external/fs/StaleAwarePersisterTest.java

```diff
@@ -0,0 +1,93 @@
+package com.nytimes.android.external.fs;
+
+import com.nytimes.android.external.fs.filesystem.FileSystem;
+import com.nytimes.android.external.store.base.impl.BarCode;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.NoSuchElementException;
+import java.util.concurrent.TimeUnit;
+
+import okio.BufferedSource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+public class StaleAwarePersisterTest {
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @Mock
+    FileSystem fileSystem;
+    @Mock
+    BufferedSource bufferedSource;
+
+    private StaleAwarePersister sourcePersister;
+    private final BarCode simple = new BarCode("type", "key");
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        sourcePersister = new StaleAwarePersister(fileSystem, TimeUnit.DAYS, 1L);
+    }
+
+    @Test
+    public void readExists() throws FileNotFoundException {
+        when(fileSystem.exists(SourcePersister.pathForBarcode(simple)))
+                .thenReturn(true);
+        when(fileSystem.read(SourcePersister.pathForBarcode(simple))).thenReturn(bufferedSource);
+
+        BufferedSource returnedValue = sourcePersister.read(simple).toBlocking().single();
+        assertThat(returnedValue).isEqualTo(bufferedSource);
```

Comment:
??? restate please. this test was primarily copied from the sourcePersister.  i didn't know if we wanted to duplicate the entire test suite to ensure that everything worked as expected, or just test the new methods i implemented.